popsycle/synthetic.py

```diff
@@ -3512,3 +3528,438 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
+                p = 0
+        if hasattr(con, '__len__'):
+            p = np.outer(p,pnfwunorm(1, con=con))
+        else:
+            p *= pnfwunorm(1, con=con)
+        return (-(1.0/np.real(special.lambertw(-np.exp(-p-1))))-1)/con
+
+    #Calculating radius values for all PBHs within r_max of the galactic center, using NFWDist functions
+    r_values = (qnfw(np.random.rand(int(num_pbh_within_r_max)) * pnfw(r_max/r_vir,con=c, logp=False), con=c)*r_vir)
+
+    #Sample PBH latitude and longitudes to get full spherical coordinates.
+    sin_lats = np.random.uniform(-1, 1, int(num_pbh_within_r_max))
+    lats=np.arcsin(sin_lats)
+    longs = np.random.uniform(0, np.pi*2, int(num_pbh_within_r_max))
+    
+    #Converting spherical galactocentric coordinates to cartesian galactocentric coordinates
+    cart = astropy.coordinates.spherical_to_cartesian(r_values, lats, longs)
+
+    #Defining galactocentric coordinate frame
+    galacto = coord.Galactocentric(x=cart[0] * units.kpc, y=cart[1] * units.kpc, z=cart[2] * units.kpc)
+
+    #Transforming from galactocentric to galactic coordinates
+    #outputs l, b, and distance in degrees.
+    galactic = galacto.transform_to(coord.Galactic(representation_type='cartesian'))
+    
+    latitude=galactic.b.deg
+    longitude=galactic.l.deg
+    #Adjusting longitude values to match the coordinate format that we need
+    longitude=np.where(longitude>180, longitude-360, longitude)   
+    
+    #Set minimum and maximum l and b for PBH mask. 
+    l_min = np.min(long_bin).values
+    l_max = np.max(long_bin).values
+    b_min = np.min(lat_bin).values
+    b_max = np.max(lat_bin).values
+
+    #Calculating distances between each latitude, longitude pair, and the center of the field of view.
+    dists = angdist(l, b, longitude, latitude)
+
+    #Masking the full PBH data to obtain just the PBHs in our field of view.
+    mask = (galactic.distance.kpc <= 2*r_max) & (dists < field_of_view_diameter)
+    data_in_field = galactic[mask]
+
+    #Obtain the radius, l, and b values for all of the PBHs in our field of view.
+    r_in_field = data_in_field.distance.kpc
+    l_in_field = data_in_field.l.deg
+    b_in_field = data_in_field.b.deg
+
+    #Converting the PBH positions from the field of view back to galactocentric for determining velocities.
+    galactic_pbh = coord.Galactic(l=l_in_field * units.deg, b=b_in_field * units.deg, distance=r_in_field * units.kpc)
+    galactic_pbh = galactic_pbh.transform_to(coord.Galactocentric(representation_type='spherical'))
+    cart_pbh = astropy.coordinates.cartesian_to_spherical(galactic_pbh.x, galactic_pbh.y, galactic_pbh.z)
+    pbh_r_galacto = cart_pbh[0]
+    
+    #Inner slope of the MW halo
+    #From Lacroix et al 2018, Figure 11 (top left panel)
+    if inner_slope == 1:
+        vel_data = pd.read_csv('data/radial_velocity_profile_steep.csv')
+    elif inner_slope == .25:
+        vel_data = pd.read_csv('data/radial_velocity_profile_shallow.csv')
+    else:
```

Comment:
What if inner_slope = 5? Or None? Or a string? It's not safe to have a catch-all "else" here. Replace this with another "elif" and then raise an exception and exit upon the else.